BLE_Button/source/main.cpp

```diff
@@ -75,13 +75,9 @@ class BatteryDemo : ble::Gap::EventHandler {
             ble::adv_interval_t(ble::millisecond_t(1000))
         );
 
-        uint8_t adv_buffer[ble::LEGACY_ADVERTISING_MAX_SIZE];

```

Comment:
I would prefer to have a buffer and a builder local to the function so we just consume stack size. The classes have been written to be used that way. We can then remove class members `_adv_data_builder` and `_adv_buffer `.

Reply:
That sounds like a good generalization exercise that demonstrates best practice. Will prepare a fresh change.